controllers/apps/transformer_cluster_backup_policy.go

```diff
@@ -360,19 +360,31 @@ func (r *clusterBackupPolicyTransformer) syncBackupPolicy(comp componentItem, ba
 	r.syncBackupPolicyTargetSpec(backupPolicy, comp)
 }
 
-func (r *clusterBackupPolicyTransformer) syncRoleLabelSelector(comp componentItem, target *dpv1alpha1.BackupTarget, role string) {
+func (r *clusterBackupPolicyTransformer) syncRoleLabelSelector(comp componentItem, target *dpv1alpha1.BackupTarget, role, alternateRole string) {
 	if len(role) == 0 || target == nil {
 		return
 	}
 	podSelector := target.PodSelector
 	if podSelector.LabelSelector == nil || podSelector.LabelSelector.MatchLabels == nil {
 		podSelector.LabelSelector = &metav1.LabelSelector{MatchLabels: map[string]string{}}
 	}
-	if r.getCompReplicas(comp) == 1 {
+	replicas := r.getCompReplicas(comp)
+	if replicas == 1 {
 		delete(podSelector.LabelSelector.MatchLabels, constant.RoleLabelKey)
 	} else if podSelector.LabelSelector.MatchLabels[constant.RoleLabelKey] == "" {
 		podSelector.LabelSelector.MatchLabels[constant.RoleLabelKey] = role
 	}
+	if len(alternateRole) == 0 {
+		return
+	}
+	if podSelector.AlternateLabelSelector == nil || podSelector.AlternateLabelSelector.MatchLabels == nil {
+		podSelector.AlternateLabelSelector = &metav1.LabelSelector{MatchLabels: map[string]string{}}
+	}
+	if replicas == 1 {
+		delete(podSelector.AlternateLabelSelector.MatchLabels, constant.RoleLabelKey)
+	} else if podSelector.AlternateLabelSelector.MatchLabels[constant.RoleLabelKey] == "" {
+		podSelector.AlternateLabelSelector.MatchLabels[constant.RoleLabelKey] = alternateRole
+	}
```

Comment:
Above code reads strange ...
The logics should be rearranged

Reply:
```
if replicas == 1 {
    delete(podSelector.LabelSelector.MatchLabels, constant.RoleLabelKey)
    if podSelector.AlternateLabelSelector != nil && podSelector.AlternateLabelSelector.MatchLabels != nil {
        delete(podSelector.AlternateLabelSelector.MatchLabels, constant.RoleLabelKey)
    }
} else {
    podSelector.LabelSelector.MatchLabels[constant.RoleLabelKey] = role
    if len(alternateRole) > 0 {
         if podSelector.AlternateLabelSelector == nil || podSelector.AlternateLabelSelector.MatchLabels == nil {
		podSelector.AlternateLabelSelector = &metav1.LabelSelector{MatchLabels: map[string]string{}}
	}
        podSelector.AlternateLabelSelector.MatchLabels[constant.RoleLabelKey] = alternateRole
    }
}

```

Reply:
I will fix it.